src/d/alexa-openwebif/source/main.d

```diff
@@ -243,49 +244,98 @@ final class OpenWebifSkill : AlexaSkill!OpenWebifSkill
 		return result;
 	}
 
-	///
-	@CustomIntent("IntentZap")
-	AlexaResult onIntentZap(AlexaEvent event, AlexaContext)
-	{
-		auto targetChannel = event.request.intent.slots["targetChannel"].value;
-
-		auto switchedTo = "nichts";
-
-		if(targetChannel.length > 0)
-		{
-			auto allservices = apiClient.getallservices();
-
-			ulong minDistance = ulong.max;
-			size_t minIndex;
-
-			foreach(i, subservice; allservices.services[0].subservices)
-			{
-				if(subservice.servicename.length < 2)
-					continue;
-
-				import std.algorithm:levenshteinDistance;
-
-				immutable dist = levenshteinDistance(subservice.servicename,targetChannel);
-				if(dist < minDistance)
-				{
-					minDistance = dist;
-					minIndex = i;
-				}
-			}
-
-			auto matchedServices = allservices.services[0].subservices[minIndex];
-
-			apiClient.zap(matchedServices.servicereference);
-
-			switchedTo = matchedServices.servicename;
-		}
-
-		AlexaResult result;
-		result.response.outputSpeech.type = AlexaOutputSpeech.Type.SSML;
-		result.response.outputSpeech.ssml = "<speak>Ich habe umgeschaltet zu: <p>"~ switchedTo ~"</p></speak>";
-
-		return result;
-	}
+ ///
+  @CustomIntent("IntentZap")
+  AlexaResult onIntentZap(AlexaEvent event, AlexaContext context)
+  {
+    auto targetChannel = event.request.intent.slots["targetChannel"].value;
+    Subservice matchedServices; 
+    auto switchedTo = "nichts";
+   
+    if(targetChannel.length > 0)
+    {
+      auto allservices = apiClient.getallservices();
+      immutable auto maxIndex = allservices.services[0].subservices.length;
+
+      if (targetChannel == "up" || targetChannel == "down")
```

Comment:
please split up this too long method at least into the up/down and the "normal" zapTo part